src/Console/Output/ErrorOutput.php

```diff
@@ -0,0 +1,138 @@
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Console\Output;
+
+use PhpCsFixer\Error\Error;
+use PhpCsFixer\Linter\LintingException;
+use Symfony\Component\Console\Formatter\OutputFormatter;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @author SpacePossum
+ *
+ * @internal
+ */
+final class ErrorOutput
+{
+    /**
+     * @var OutputInterface
+     */
+    private $output;
+
+    /**
+     * @var bool
+     */
+    private $isDecorated;
+
+    public function __construct(OutputInterface $output)
+    {
+        $this->output = $output;
+        $this->isDecorated = $output->isDecorated();
+    }
+
+    /**
+     * @param string  $process
+     * @param Error[] $errors
+     */
+    public function listErrors($process, array $errors)
+    {
+        $this->output->writeln(array('', sprintf(
+            'Files that were not fixed due to errors reported during %s:',
+            $process
+        )));
+
+        $showDetails = $this->output->getVerbosity() >= OutputInterface::VERBOSITY_VERBOSE;
+        $showTrace = $this->output->getVerbosity() >= OutputInterface::VERBOSITY_VERY_VERBOSE;
+        foreach ($errors as $i => $error) {
+            $this->output->writeln(sprintf('%4d) %s', $i + 1, $error->getFilePath()));
+            if ($showDetails) {
+                $e = $error->getSource();
+                if (null !== $e) {
+                    $class = sprintf('[%s]', get_class($e));
+                    $message = $e->getMessage();
+                    $code = $e->getCode();
+                    if (0 !== $code) {
+                        $message .= " ($code)";
+                    }
+
+                    $length = max(strlen($class), strlen($message));
+                    $lines = array(
+                        '',
+                        $class,
+                        $message,
+                        '',
+                    );
+
+                    $this->output->writeln('');
+
+                    foreach ($lines as $line) {
+                        if (strlen($line) < $length) {
+                            $line .= str_repeat(' ', $length - strlen($line));
+                        }
+
+                        $this->output->writeln(sprintf('      <error>  %s  </error>', $this->prepareOutput($line)));
+                    }
+
+                    if ($showTrace && !$e instanceof LintingException) { // stack trace of lint exception is of no interest
+                        $this->output->writeln('');
+                        $stackTrace = $e->getTrace();
+                        $stackTraceCount = count($stackTrace);
+                        foreach ($stackTrace as $trace) {
+                            if ($stackTraceCount > 4 && isset($trace['class'], $trace['function']) && 'Symfony\Component\Console\Command\Command' === $trace['class'] && 'run' === $trace['function']) {
```

Comment:
I would remove `$stackTraceCount > 4 && ` part.
If error is inside our code, it's nice to have full stack.